packages/stylelint-plugin/src/rules/lib/tokensInFunctions.ts

```diff
@@ -0,0 +1,202 @@
+    if (isNaN(parsed)) {
+      return "NaN"
+    } else {
+      return parsed
+    }
+  }
+}
+
+/**
+ * Given a string like "50%", "40%", "30", "15" etc (you don't have to pass in the percentage sign), return a decimal like 0.5, 0.4, 0.3 etc
+ */
+const parsePercentage = (value: string): number | "NaN" => {
+  // Add-alpha can be used without a percentage sign, that won't fly.
+  // Fix the percentage sign, then convert to a decimal for usage in RGBA
+  const parsed = parseInt(value.replace(/\%/, ""), 10)
+  if (isNaN(parsed)) {
+    return "NaN"
+  }
+  return parsed / 100
+}
+
+/*
+  Return a replacement rgb variable if required.
+  Will also migrate a deprecated variable within an alpha manipulation function
+*/
+const getReplacementRgbTriple = (
+  kaizenToken: KaizenToken
+):
+  | { replacement: CurrentKaizenToken }
+  /* For example, the token is `$kz-deprecated-color-lapis`, and there is no replacement. */
+  | "CantFindReplacementForDeprecatedToken"
+  /* Protecting against mistakes where there is no matching -rgb token */
+  | "NoRGBTripleReplacement"
+  /* The token is already an RGB triple, or is not a CSS variable. */
+  | "NoReplacementNeeded" => {
+  // We'll decide on what this variable should be using a series of rules, then use it as the subject for making replacements
+  let kaizenTokenSubject: CurrentKaizenToken
+
+  const replacement =
+    kaizenTokensByName[deprecatedTokenReplacements.get(kaizenToken.name) || ""]
+
+  if (replacement) {
+    kaizenTokenSubject = replacement
+  } else if (!kaizenToken.deprecated) {
+    kaizenTokenSubject = kaizenToken
+  } else {
+    // Just bail if we reach here, because we couldn't decide on a kaizen token to use.
+    return "CantFindReplacementForDeprecatedToken"
+  }
+
+  // If the value is not like `123, 123, 123`, AND it's a CSS variable, it's not valid
+  if (
+    kaizenTokenSubject.cssVariable &&
+    kaizenTokenSubject.cssVariable.fallback &&
+    !isRgbTriple(kaizenTokenSubject.cssVariable.fallback)
+  ) {
+    const fixedVariable = kaizenTokensByName[kaizenTokenSubject.name + "-rgb"]
+    if (!fixedVariable) {
+      return "NoRGBTripleReplacement"
+    }
+    return { replacement: fixedVariable }
+  }
+
+  return "NoReplacementNeeded"
+}
+
+// Does the value look like `123, 123, 123`? Used for determining whether an rgba function is used with the correct variable
+const isRgbTriple = (value: string) =>
+  /^\s*\d{1,3}\s*,\s*\d{1,3}\s*,\s*\d{1,3}\s*$/.test(value)
+
+/**
+ * Given a CSS declaration value, find any alpha modifying functions like add-alpha, transparentize, or rgba, then normalise them to either rgba or rgb,
+ * and ensure that a RGB triple Kaizen token is being used, e.g. `$color-purple-100-rgb` rather than `$color-purple-100`.
+ * It also knows how to migrate from a deprecated variable within a function, e.g. `$kz-color-wisteria-100` -> `$color-purple-100-rgb`
+ */
+export const fixAlphaModificationFunctions = (
```

Comment:
This is the only export... should we call the file `fixAlphaModificationFunctions.ts`?

Then it avoids becoming a grab-bag utils file